http-clients/src/main/java/com/palantir/remoting3/clients/ClientConfigurations.java

```diff
@@ -114,4 +116,13 @@ private static ProxySelector fixedProxySelectorFor(Proxy proxy) {
             public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {}
         };
     }
+
+    private static Integer getMaxNumRetries(int numUris) {
```

Comment:
not a fan of silently overwriting the user-supplied config. would prefer no limits.

Reply:
This is not overwriting any user-supplied config though. This is only used if the maxNumRetries is not provided right?